src/platform/EFR32/DiagnosticDataProviderImpl.cpp

```diff
@@ -287,6 +287,8 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface **
         }
         else
         {
+            // Setting the type to EMBER_ZCL_INTERFACE_TYPE_WI_FI since GetInterfaceType returns CHIP_ERROR_NOT_IMPLEMENTED for LWIP
+            ifp->type = EMBER_ZCL_INTERFACE_TYPE_WI_FI;
```

Comment:
The error condition leading to forcing WiFi type is not the right way to fix it. Either `GetInterfaceType` needs to be implemented in inet_Interface.cpp to return real determined values, or don't use that function in your if here and implement/define a way to differentiate what type your interface is in DiagnosticDataProviderImpl.cpp

Reply:
Hmm will removing the switch case from there and setting the type to WIFI will be better. 
Since GetInterfaceType maybe implemented in the future so thought of better keeping this for now. Any suggestion what will be better